hgweb/command-runner.sh

```diff
@@ -44,6 +44,10 @@ echo ""
 
 # Run the hg command, simply output to stdout
 first_char=$(echo $project_code | cut -c1)
+# Ensure NFS cache is refreshed in case project repo changed in another pod (e.g., project reset)
+ls /var/hg/repos/$first_char/$project_code/.hg >/dev/null 2>/dev/null  # Don't need output; this is enough to refresh NFS dir cache
+# Sometimes invalidatedircache is called after deleting a project, so the cd would fail. So exit fast in that case.
+[ "x$command_name" = "xinvalidatedircache" ] && exit 0
```

Comment:
why prefix with x?

Reply:
It's a standard approach to prevent an error message if trying to compare an empty string. (`[ "" = "foo" ]` would be invalid syntax, but `[ "x" = "xfoo" ]` would be legal.) See the `=` example at https://tldp.org/LDP/abs/html/ops.html for more. In this case, we already check `$command_name` against a short list of valid strings so it's not possible for it to be empty, but it's still best practice to prefix with x when using the `=` operator, so I did it.